lib/context_adapter_server.js

```diff
@@ -274,6 +319,11 @@ function requestHandler(request, reply) {
 
   if (caComm.isPollingUpdateContext(request)) {
     respond2Request(request, reply);
+  } else if (caComm.getCellIdDescriptors(request).length) {
+    async.parallel([
+      respond2Request.bind(null, request, reply),
+      processCellIdRequest.bind(null, request)
```

Comment:
Imagine an asteroid impacts The Earth in the precise moment that line is run... That would hurt too... :p Let's say it is safe to run both functions in parallel according to what they do inside instead (since they are not 100% independent as far as they have access to the same object as you point out)... ;)